operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -698,17 +729,56 @@ private void addIncludesSearchParameters(String compartmentMemberType, Multivalu
         String simplifiedParameterName = compartmentMemberType + ":" + code;
-        if (allowedIncludes == null || allowedIncludes.contains(parameterName)
-                || allowedIncludes.contains(simplifiedParameterName)) {
+        if ((extraResources != null && extraResources.contains(subResourceType.value())) &&
+                (allowedIncludes == null || allowedIncludes.contains(parameterName)
+                || allowedIncludes.contains(simplifiedParameterName))) {
             searchParameters.add("_include", parameterName);
         } else {
             if (LOG.isLoggable(Level.FINE)) {
                 LOG.fine("Filtering out searchParameter because it is not supported by the server config: " + parameterName);
             }
         }
     }
+    
+    private void readsOfAdditionalAssociatedResources(String compartmentMemberType, List<Entry> newEntries,
+        List<Entry> allEntries, List<String> resourceIds, MultivaluedMap<String, String> searchParameters, 
+        FHIRResourceHelpers resourceHelper, List<String> extraResources) throws Exception {
+        if (extraResources == null || extraResources.isEmpty()) {
+            return;
+        }
+        List<Resource> resourceList = new ArrayList<Resource>();
+        for (Entry entry: newEntries) {
+            // Look up entries by reference
+            List<Reference> references = ReferenceFinder.getReferences(entry.getResource());
+            for (Reference reference: references) {
+                ReferenceValue referenceValue = ReferenceUtil.createReferenceValueFrom(reference, compartmentMemberType);
+                if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_RELATIVE)) {
+                    String externalId = reference.getReference().as(com.ibm.fhir.model.type.String.class).getValue();
+                    String[] splitString = externalId.split("/");
+                    String type = splitString[0];
+                    if (!resourceIds.contains(externalId) && extraResources.contains(type) && (type.equals(ResourceType.Value.LOCATION.value()) || type.equals(ResourceType.Value.MEDICATION.value())
+                            || type.equals(ResourceType.Value.ORGANIZATION.value()) || type.equals(ResourceType.Value.PRACTITIONER.value()))) {
+                        resourceList.add(resourceHelper.doRead(type, splitString[1], false, false, null, null).getResource());
+                        resourceIds.add(externalId);
+                    }
+                }
+            }
+        }
+
+        // Transform resources into Entries
+        for (Resource resource: resourceList) {
+            Bundle.Entry.Builder entryBuilder = Bundle.Entry.builder();
+           
+            FHIRRequestContext requestContext = FHIRRequestContext.get();
+            String url = requestContext.getOriginalRequestUri();
+            url = url.substring(0, url.indexOf("/Patient/"));
```

Comment:
Let's do this one time. 

It's slightly better to do it on one line in Java8 byte code.  The compiler does an optimization on the URL. 
This is only needed one time per execution (e.g. call to EverythingOperation.invvoke

Reply:
Done.